crates/kitsune_p2p/dht_arc/src/dht_arc.rs

```diff
@@ -300,6 +300,32 @@ impl DhtArcRange<DhtLocation> {
         matches!(self, Self::Bounded(_, _))
     }
 
+    /// Get the min distance to a location.
+    /// Zero if Full, u32::MAX if Empty.
+    pub fn dist(&self, tgt: u32) -> u32 {
+        match self {
+            DhtArcRange::Empty => u32::MAX,
+            DhtArcRange::Full => 0,
+            DhtArcRange::Bounded(start, end) => {
+                let start = u32::from(*start);
+                let end = u32::from(*end);
+                if start < end {
+                    if tgt >= start && tgt <= end {
+                        0
+                    } else if tgt < start {
+                        std::cmp::min(start - tgt, (u32::MAX - end) + tgt)
+                    } else {
+                        std::cmp::min(tgt - end, (u32::MAX - tgt) + start)
+                    }
+                } else if tgt <= end || tgt >= start {
+                    0
+                } else {
+                    std::cmp::min(tgt - end, start - tgt)
+                }
+            }
+        }
+    }
```

Comment:
Would be nice to have a unit test for this

Reply:
There is a unit test validating this. Although I copied the code for this function out of that unit test, so, of course it's going to pass : )